microcosm/loaders.py

```diff
@@ -95,23 +95,28 @@ def _load_from_environ(metadata, value_func=None):
 
     prefix = metadata.name.upper().replace("-", "_").split("_")
 
-    def matches_key(key_parts):
+    def matches_app(key_parts):
```

Comment:
I think this condition is really, is the env-var we are processing namespaced to the current app.